merge-rfc.sh

```diff
@@ -48,36 +94,66 @@ while getopts ":i:n" opt; do
   esac
 done
 
-if [[ $no_issues = false && $issues == "N/A" ]]; then
-  echo -e "ERROR! No issues were provided. Are you sure there are no issues that should be linked?"
-  echo -e "ERROR! Either -i or -n is required\n"
-  usage
+CURRENT_BRANCH=$(git branch --show-current)
+if [[ "${CURRENT_BRANCH}" != "${MAIN_BRANCH}" ]]; then
+  echo -e "ERROR! Expected current branch to be '${MAIN_BRANCH}', currently in '${CURRENT_BRANCH}'!";
+  exit 1;
 fi
 
-shift $(($OPTIND - 1))
-if [[ $# != 1 ]]; then
+####
+# TASK
+#
+
+RFC_ID=$(generate_id)
+echo "> Generated RFC number: ${RFC_ID}"
+
+echo "> Creating issues..."
+export GITHUB_TOKEN
+
+ISSUES_GEN_CLI="node .github/workflows/actions/issues-generation/dist/index.js"
+$ISSUES_GEN_CLI create --pr "${OWNER}/${REPO}#${PR_NUMBER}" --bot $BOT_USERNAME  --prepend "[RFC #${RFC_ID}] "
+ISSUES_TO_LINK=$($ISSUES_GEN_CLI list --pr "${OWNER}/${REPO}#${PR_NUMBER}" --bot $BOT_USERNAME --json | jq -r '[.[] | select(.num) | .repo + "#" + (.num|tostring) ] | join(" ")')
+
+for ISSUE in ${ISSUES_TO_LINK}; do
+  if [[ $ISSUES_TEXT == "N/A" ]]; then
+    ISSUES_TEXT=$(link_issue "$ISSUE")
+  else
+    ISSUES_TEXT+=", $(link_issue "$ISSUE")"
+  fi
+done
+
+
+if [[ $NO_ISSUES = false && $ISSUES_TEXT == "N/A" ]]; then
+  echo -e "ERROR! No issues were provided. Are you sure there are no issues that should be linked?"
+  echo -e "ERROR! Either -i or -n is required\n"
   usage
 fi
 
+echo "> Pulling latest changes...."
 git pull origin --rebase
 
-PR="${1}"
-ID=$(id)
+echo "> Merging PR#${PR_NUMBER} as rfc-${RFC_ID}"
+git fetch origin "pull/${PR_NUMBER}/head:rfc-${RFC_ID}"
+git merge "rfc-${RFC_ID}" --signoff --no-edit --no-ff
+git branch -d "rfc-${RFC_ID}"
 
-echo "Merging PR ${PR} as rfc-${ID}"
+SOURCE_DOC=$(find text -maxdepth 1 -name '0000-*')
+TARGET_DOC=${SOURCE_DOC//0000/${RFC_ID}}
 
-git fetch origin "pull/${PR}/head:rfc-${ID}"
-git merge "rfc-${ID}" --signoff --no-edit --no-ff
-git branch -d "rfc-${ID}"
+echo "> Updating document: ${SOURCE_DOC}"
+SEDOPTION="-i"
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  SEDOPTION="-i ''"
+fi
+sed $SEDOPTION "s|- RFC Pull Request:.*|- RFC Pull Request: [${REPO}#${PR_NUMBER}](https://github.com/${OWNER}/${REPO}/pull/${PR_NUMBER})|" "${SOURCE_DOC}"
+sed $SEDOPTION "s|- CNB Issue:.*|- CNB Issue: $ISSUES_TEXT|" "${SOURCE_DOC}"
 
-SOURCE=$(find text -depth 1 -name '0000-*')
-TARGET=${SOURCE//0000/$(printf "%04d" "${ID}")}
+echo "> Moving ${SOURCE_DOC} to ${TARGET_DOC}..."
+git mv "${SOURCE_DOC}" "${TARGET_DOC}"
+git add "${TARGET_DOC}"
 
-sed -i '' "s|- RFC Pull Request:.*|- RFC Pull Request: [rfcs#${PR}](https://github.com/buildpacks/rfcs/pull/${PR})|" "${SOURCE}"
-sed -i '' "s|- CNB Issue:.*|- CNB Issue: $issues|" "${SOURCE}"
-git mv "${SOURCE}" "${TARGET}"
-git add "${TARGET}"
-git commit --signoff --message "RFC ${ID}
+echo "> Committing..."
+git commit --signoff --message "RFC ${RFC_ID}
```

Comment:
What's the configured git user here for the `signoff` and `commit`?

Reply:
This would be the current user running `./merge-rfc.sh`. You, me, anyone. :)